oemof/tools/helpers.py

```diff
@@ -57,3 +59,24 @@ def flatten(d, parent_key='', sep='_'):
         else:
             items.append((new_key, v))
     return dict(items)
+
+
+def calculate_timeincrement(timeindex):
+    """
+    Calculates timeincrement for `timeindex`
+    """
+    if isinstance(timeindex, pd.DatetimeIndex):
+        if len(set(timeindex)) != len(timeindex):
+            raise IndexError("No equal DatetimeIndex allowed!")
+        timeindex = timeindex.to_series()
+        timeindex_sorted = timeindex.sort_values()
+        timeincrement = timeindex_sorted.diff().dropna()
+        timeincrement_sec = timeincrement.map(dt.timedelta.total_seconds)
+        timeincrement_hourly = list(timeincrement_sec.map(
+                                    lambda x: x/3600))
+        timeincrement_hourly.append(1.0)
```

Comment:
I found a small bug. I would like to change this line to 
```python 
timeincrement_hourly.insert(0, 1.0)
```
the pandas.Series.diff() function results to a `timeincrement` with length-1 compared to the `timeindex`, because the first value will be missed out.
The output of
```python
timeidxH = pd.date_range('1/1/2017', periods=3, freq='H')
timeidx2H = pd.date_range("1/1/2017 04:00:00", periods = 2, freq="2H")
timeindex = timeidxH.append([timeidx2H])
print(timeindex)
print(timeindex.diff()
```
for `timeindex` is

2017-01-01 00:00:00   2017-01-01 00:00:00
2017-01-01 01:00:00   2017-01-01 01:00:00
2017-01-01 02:00:00   2017-01-01 02:00:00
2017-01-01 04:00:00   2017-01-01 04:00:00
2017-01-01 06:00:00   2017-01-01 06:00:00

and for `timeincrement` is 

2017-01-01 00:00:00        NaT
2017-01-01 01:00:00   01:00:00
2017-01-01 02:00:00   01:00:00
2017-01-01 04:00:00   02:00:00
2017-01-01 06:00:00   02:00:00

So it would make more sense of inserting the missing timeincrement value on the first place, instead on the last one.

I will change it with the following commit.

Reply:
Sounds reasonable.